lib/cstruct_cap.mli

```diff
@@ -0,0 +1,88 @@
+type 'a rd = < rd: unit; .. > as 'a
+type 'a wr = < wr: unit; .. > as 'a
+type 'a aligned = < aligned: unit; .. > as 'a
+
+type 'a t = private Cstruct_core.t
+
+type rdwr =  < rd: unit; wr: unit; >
+type ro = < rd: unit; >
+type wo = < wr: unit; >
+
+type uint8 = int
+type uint16 = int
+type uint32 = int32
+type uint64 = int64
+
+val create : int -> rdwr t
+val ro : 'a rd t -> ro t
+val wo : 'a wr t -> wo t
+
+val of_string : ?allocator:(int -> rdwr t) -> ?off:int -> ?len:int -> string -> rdwr t
+val of_bytes : ?allocator:(int -> rdwr t) -> ?off:int -> ?len:int -> bytes -> rdwr t
+val of_hex : string -> rdwr t
+
+val equal : 'a rd t -> 'b rd t -> bool
+val compare : 'a rd t -> 'b rd t -> int
+
+val check_bounds : 'a rd t -> int -> bool
+val check_alignment : 'a rd t -> int -> bool
+
+val get_char : 'a rd t -> int -> char
+val get_uint8 : 'a rd t -> int -> uint8
+val set_char : 'a wr t -> int -> char -> unit
+val set_uint8 : 'a wr t -> int -> uint8 -> unit
+
+val sub : 'a rd t -> int -> int -> 'a rd t
+val shift : 'a rd t -> int -> 'a rd t
+val copy : 'a rd t -> int -> int -> string
+
+val blit : 'a rd t -> int -> 'b wr t -> int -> int -> unit
+val blit_from_string : string -> int -> 'a wr t -> int -> int -> unit
+val blit_from_bytes : bytes -> int -> 'a wr t -> int -> int -> unit
+val blit_to_bytes : 'a rd t -> int -> bytes -> int -> int -> unit
+
+val memset : 'a wr t -> int -> unit
+
+val len : 'a t -> int
+val set_len : 'a wr t -> int -> 'a wr t
+val add_len : 'a wr t -> int -> 'a wr t
```

Comment:
wow, I've not seen this before, and am not sure anyone is using it (GitHub search showed test cases)

Reply:
Yeah, I think we should remove these. Searching my local disk, I found that mirage-block-xen uses `add_len`:

https://github.com/mirage/mirage-block-xen/blob/master/lib/block_request.ml#L61

It would probably be better to move this operation into cstruct itself, e.g.

```ocaml
val coalesce : t -> t -> t option
(** [coalesce a b] is a cstruct that covers [a] and [b] if they belong to the same
    underlying array and [b] follows [a] directly in memory. *)
```

Reply:
Waiting to merge #245